apache/incubator-gluten
# [VL] Enable GlutenParqutRowIndexSuite for Spark 3.4/3.5

## What changes were proposed in this pull request?

The PR use to enable GlutenParquetRowIndexSuite. 
The suite do below steps to run tests:
1. Run parquet write api `df.write` to generate specific parquet test data file with specific option as block size, page size, etc.
2. Check if the generated parquet data file meets the requirements as the block size is expected or not.
3. If the generated parquet files are ok for use, then run all `row index generation xx` and `invalid row index type xxx` uts.

However, gluten parquet native write currenrly only does not support customize the `block_size`, `page_size`, that means 
the generated parquet file would always has hard-code row group number / page size per file which causes the tests run fail in parquet file generation stage. 

Checking here for the default block_size, page_size use in velox parquet write.

https://github.com/facebookincubator/velox/blob/b29d93374d8e75a595c85550428eb008a371d456/velox/dwio/parquet/writer/Writer.h#L90C3-L102C74


https://github.com/facebookincubator/velox/blob/main/velox/dwio/parquet/writer/Writer.cpp#L420C3-L420C52

The PR re-write the suite to disable the native write to always generate the test parquet file content with vinaila spark and some addition check to fix the uts.


(Fixes: \#5309)

## How was this patch tested?

manually test / local run

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)


(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
